apps/editing-toolkit/editing-toolkit-plugin/event-countdown-block/blocks/src/edit.js

```diff
@@ -5,14 +5,48 @@
 import { Button, DateTimePicker, Dropdown, Placeholder } from '@wordpress/components';
 import { dateI18n, __experimentalGetSettings } from '@wordpress/date';
 import { __ } from '@wordpress/i18n';
+import moment from 'moment';
 import { EventCountdownIcon } from './icon';
 
+const TIMEZONELESS_FORMAT = 'YYYY-MM-DDTHH:mm:ss';
+
+/**
+ * Assigns timezone to a date without altering it
+ *
+ * @param {string} date a date in YYYY-MM-DDTHH:mm:ss format
+ * @param {number} offset the offset in hours
+ * @returns a moment instance
+ */
+function assignTimezone( date, offset, format = TIMEZONELESS_FORMAT ) {
+	return moment( date, format ).utcOffset( offset * 60, true );
+}
+
 const edit = ( { attributes, setAttributes, className } ) => {
 	const settings = __experimentalGetSettings();
 
 	let label = __( 'Choose Date', 'full-site-editing' );
-	if ( attributes.eventDate ) {
-		label = dateI18n( settings.formats.datetimeAbbreviated, attributes.eventDate );
+	let eventDate;
+
+	if ( attributes.eventTimestamp ) {
+		label = dateI18n(
+			settings.formats.datetimeAbbreviated,
+			new Date( attributes.eventTimestamp * 1000 )
+		);
+
+		// the DateTimePicker requires the date to be in this format
+		// we offset the date by the site timezone settings to counteract the Datepicker automatic adjustment to the client-side timezone
+		eventDate = moment( attributes.eventTimestamp * 1000 )
+			.utcOffset( settings.timezone.offset * 60 )
+			.format( TIMEZONELESS_FORMAT );
+	} else if ( attributes.eventDate ) {
+		// backwards compatibility
+		const siteTimeZoneAdjustedTime = assignTimezone(
+			attributes.eventDate,
+			Number.parseFloat( settings.timezone.offset ) // offset can be a string if a manual timezone is selected
```

Comment:
By manual timezone, you are referring to the "Site Settings > Site Timezone > Manual Offsets" dropdown selection right?

![2021-12-14_11-59-02](https://user-images.githubusercontent.com/1287077/145976483-b4b4d3b7-cce5-4e7b-bed3-d2c9b2193a70.png)

Reply:
This is now verified. Thanks for parsing it. :)

![image](https://user-images.githubusercontent.com/1287077/145987362-b3ceb55d-beaa-442b-8bce-cbac20c1f070.png)

![image](https://user-images.githubusercontent.com/1287077/145987374-453aafa0-fa34-4b14-be8e-d70a9c0e7324.png)